tests/specs/test_yaml_file.py

```diff
@@ -0,0 +1,24 @@
+import unittest
+try:
+    from unittest import mock
+except ImportError:
+    import mock
+
+from conda_env import env
+from conda_env.specs.yaml_file import YamlFileSpec
+
+
+class TestYAMLFile(unittest.TestCase):
+    def test_no_environment_file(self):
+        spec = YamlFileSpec(name=None, filename='not-a-file')
+        self.assertEqual(spec.can_handle(), False)
+
+    def test_environment_file_exist(self):
+        with mock.patch.object(env, 'from_file', return_value={}):
+            spec = YamlFileSpec(name=None, filename='environment.yaml')
+            self.assertTrue(spec.can_handle())
+
+    def test_get_environment(self):
+        with mock.patch.object(env, 'from_file', return_value={}):
```

Comment:
I love being able to read 3 - 5 line test methods when doing a review.  Really helps me get a handle on what's going on.  Nice work!

This is me showing my TDD habits so feel free to take with a grain of salt.  A good habit to get into with assertions like this is to test with some sort of random data.  For this particular function, there are two important things:

* `from_file` is given the `filename` that's provided
* whatever is returned from `from_file` is returned to the user

Can you add a test that verifies that is provided to the patched `from_file` and ensure that the value passed in as `filename` is passed along?  Having a random filename can help you verify that whatever is provided to `YamlFileSpec` is what actually gets passed in to `from_file`.

For the second test, I could write out `YamlFileSpec` to do something like this and this test would pass since `{}` is hard coded:

```python
class YamlFileSpec(object):
    def __init__(self, **kwargs):
        self.environment = {}
```

Adding a `random_return = random.randint(100, 200)` and using that value for `return_value` of the mock object means that it verifies that the randomly generated value that the mock object returns is what actually got returned.

---

Quick digression: you have to be very careful to find the right balance with mock objects.  Too many can be a sign of something going wrong (too complex, too many dependencies, etc).  You can also end up with tests that are too isolated from the rest of the system so the tests still pass but the code doesn't work like it's expected to.

`from_file` [is tested][1] (note to self, I should have written more) so given a valid `filename` you know that it will return what you're looking for.  In that case, you don't need to test `from_file`, so mocking it and making some assumptions about it is the sweet spot for mocks.

[1]: https://github.com/conda/conda-env/blob/develop/tests/test_env.py#L31-L44